src/heap/heap.ts

```diff
@@ -0,0 +1,76 @@
+export default class Heap<T> {
+  protected memory: T[] = [];
+
+  constructor(protected CompareFunction: (a: T, b: T) => number) {
```

Comment:
only classes (functions to construct objects) and types should use uppercase names. this should be compareFuntion